File: ios/MenuMaker/Data/Models/MenuModels.swift
import Foundation

// MARK: - Menu Models

struct Menu: Codable, Identifiable {
    let id: String
    let businessId: String
    let name: String
    let description: String?
    let isActive: Bool
    let displayOrder: Int
    let createdAt: String
    let updatedAt: String

    var displayDescription: String {
        description ?? "No description"
    }

    var formattedCreatedDate: String {
        guard let date = ISO8601DateFormatter().date(from: createdAt) else {
            return createdAt
        }

        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        return formatter.string(from: date)
    }
}

struct MenuResponse: Decodable {
    let success: Bool
    let data: MenuData
}

struct MenuData: Decodable {
    let menu: Menu
}

struct MenuListResponse: Decodable {
    let success: Bool
    let data: MenuListData
}

struct MenuListData: Decodable {
    let menus: [Menu]
}

struct CreateMenuRequest: Encodable {
    let businessId: String
    let name: String
    let description: String?
    let isActive: Bool
    let displayOrder: Int
}

struct UpdateMenuRequest: Encodable {
    let name: String?
    let description: String?
    let isActive: Bool?
    let displayOrder: Int?
}

File: android/app/src/main/kotlin/com/menumaker/data/remote/models/MenuModels.kt
package com.menumaker.data.remote.models

import com.google.gson.annotations.SerializedName

/**
 * Data Transfer Object for Menu
 * Represents a menu within a business/seller's offerings
 */
data class MenuDto(
    @SerializedName("id") val id: String,
    @SerializedName("business_id") val businessId: String,
    @SerializedName("name") val name: String,
    @SerializedName("description") val description: String?,
    @SerializedName("is_active") val isActive: Boolean,
    @SerializedName("display_order") val displayOrder: Int,
    @SerializedName("created_at") val createdAt: String,
    @SerializedName("updated_at") val updatedAt: String
)

/**
 * API Response for a single menu
 */
data class MenuResponse(
    @SerializedName("success") val success: Boolean,
    @SerializedName("data") val data: MenuData
)

data class MenuData(
    @SerializedName("menu") val menu: MenuDto
)

/**
 * API Response for a list of menus
 */
data class MenuListResponse(
    @SerializedName("success") val success: Boolean,
    @SerializedName("data") val data: MenuListData
)

data class MenuListData(
    @SerializedName("menus") val menus: List<MenuDto>
)

/**
 * Request to create a new menu
 */
data class CreateMenuRequest(
    @SerializedName("business_id") val businessId: String,
    @SerializedName("name") val name: String,
    @SerializedName("description") val description: String?,
    @SerializedName("is_active") val isActive: Boolean,
    @SerializedName("display_order") val displayOrder: Int
)

/**
 * Request to update an existing menu
 */
data class UpdateMenuRequest(
    @SerializedName("name") val name: String?,
    @SerializedName("description") val description: String?,
    @SerializedName("is_active") val isActive: Boolean?,
    @SerializedName("display_order") val displayOrder: Int?
)
